### PR TITLE
[openstack_blockstorage_volume_v3] Create cinder volume from backup

### PR DESCRIPTION
**Pull request by ZdenekPesek**
Close: https://github.com/terraform-provider-openstack/terraform-provider-openstack/issues/1636

### CODE REVIEW

File: openstack/resource_openstack_blockstorage_volume_v3.go [hunk 1]
```diff
@@ -93,6 +93,11 @@ func resourceBlockStorageVolumeV3() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"backup_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
```
Comment (nikParasyr) [inline openstack/resource_openstack_blockstorage_volume_v3.go:99]: Also `ForceNew: true,` should be set

File: openstack/resource_openstack_blockstorage_volume_v3.go [hunk 2]
```diff
@@ -193,6 +198,7 @@ func resourceBlockStorageVolumeV3Create(ctx context.Context, d *schema.ResourceD
 		ConsistencyGroupID: d.Get("consistency_group_id").(string),
 		Description:        d.Get("description").(string),
 		ImageID:            d.Get("image_id").(string),
+		BackupID:           d.Get("backup_id").(string),
```
Comment (nikParasyr) [inline openstack/resource_openstack_blockstorage_volume_v3.go:201]: When the user has defined `backup_id` you need to set the blockstorage client microversion to 3.47

**Review by nikParasyr — CHANGES_REQUESTED** [review]
@ZdenekPesek Hello. thanks for opening this.

Have left a couple of comments.

Moreover there should be a test case for this and also docs should be updated.

File: openstack/resource_openstack_blockstorage_volume_v3.go (hunk 1 above)
Comment (kayrus) [reply openstack/resource_openstack_blockstorage_volume_v3.go:99]: I think there should be a conflict set for all sources:
* `snapshot_id`
* `source_vol_id`
* `image_id`
* `backup_id`

**Pull request closed without merging**